### DISCUSSION TITLE
How to publish native messaging extension

### DISCUSSION THREAD

**Discussion by briceandre** [microsoft/MicrosoftEdge-Extensions#99] [Q&A] [answered]
Hello,

I am trying to publish an extension which uses native messaging to communicate with native windows program. 

On Mac, it was possible to register an application in the AppleStore so that the web-site, when the extension is not installed, automatically redirects to the proper appstore entry, and when the user installs the app, everything is installed automatically (the native mac application, the safari extension, and the binding between both).

Is it possible to achieve something equivalent in windows for Edge browser ? 

From what I read, if I publish my application from the edge store, I will be able to publish the extension itself, but not the native windows application it communicates with. 

I was guessing that maybe the solution would be to publish everything from the windows store, but I do not find any explanation on how to publish an extension from there, as well as to automate the binding between the native application and the extension.

Any help or link to dedicated documentation would be very appreciated.

Many thanks,
Brice

**Comment by Bhuvana-Priya** [comment, marked as answer]
Hi @briceandre ,

Here is a link to our documentation which has all the details about native messaging: https://learn.microsoft.com/en-us/microsoft-edge/extensions-chromium/developer-guide/native-messaging?tabs=v3%2Cwindows

Please let us know for any further queries.
